### DISCUSSION TITLE
Would appreciate some more info concerning systemd

### DISCUSSION THREAD

**Discussion by CaptainFrosty** [WhitewaterFoundry/Pengwin#764] [Q&A] [answered]
Been using Pengwin a while now, thanks for your great software that makes WSL Linux easier to deal with and is really one of a kind. I just would like to make sure my understandings are correct and get some elaboration concerning systemd and WSL if possible. It is my understanding, though I could be outdated, that the Microsoft-produced WSL kernel does not allow any process except its own init process to be PID 1, hence no systemd support. 

Pengwin still installs various systemd packages from the Debian repos, I noticed. Are these just dependencies something else needs, or some such, despite a nonfunctional systemd? 

I am considering buying Pengwin Enterprise for the sake of simplicity, since I use a RHEL-based distro on my server. I believe it advertises full systemd support. How was this accomplished with the custom kernel? I know workarounds exist, but the ones I have looked at seemed alot like hacks and kludges that might be dicey. Just curious how Pengwin Enterprise implements systemd before I buy it. 

That's about it. Any info is appreciated.

**Comment by crramirez** [comment]
Ok, give me some time to answer your question appropriately. Meanwhile, you can read this: https://www.whitewaterfoundry.com/blog/2022/8/26/announcing-built-in-systemd-support-in-pengwin-enterprise the part that begins with "Traditionally bringing SystemD to WSL implied using workarounds..." but there is more to explain that is not said in this article because things have changed

**Comment by crramirez** [comment, marked as answer]
Pengwin and Pengwin Enterprise have built-in systemd support. We are using a port of this script: https://github.com/diddledani/one-script-wsl2-systemd/blob/main/src/00-wsl2-systemd.sh and what it does is to execute a subsystem as systemD as process id 1 and then everything bellow this.

To enable SystemD in Pengwin, you can do it via pengwin setup, in the Services section.
In the case of Pengwin Enterprise, it has a shortcut that finishes in (SystemD) in Windows Terminal and Start menu.

But recently, Microsoft implemented its own SystemD, and Pengwin and Pengwin Enterprise are compatible with it: https://devblogs.microsoft.com/commandline/systemd-support-is-now-available-in-wsl/

So, we don't have a custom kernel, and SystemD is implemented as one of these hacks. But works.

You can buy it with confidence; if it doesn't work for you, simply write us an email and you'll get a full refund, via PayPal.

**Comment by kuttysoftmy** [comment]
well pengwin normal one not support custom kernel

**Reply by crramirez** [reply]
mmm? is it a question?
